src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -23,8 +23,23 @@ public MySqlConnection(string connectionString)
 
 		public new MySqlTransaction BeginTransaction() => (MySqlTransaction) base.BeginTransaction();
 
+		public async Task<MySqlTransaction> BeginTransactionAsync(CancellationToken cancellationToken = default(CancellationToken))
+		{
+			return await BeginDbTransactionAsync(IsolationLevel.Unspecified, cancellationToken).ConfigureAwait(false) as MySqlTransaction;
+		}
+
+		public async Task<MySqlTransaction> BeginTransactionAsync(IsolationLevel isolationLevel, CancellationToken cancellationToken = default(CancellationToken))
+		{
+			return await BeginDbTransactionAsync(isolationLevel, cancellationToken).ConfigureAwait(false) as MySqlTransaction;
+		}
+
 		protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel)
 		{
+			return BeginDbTransactionAsync(isolationLevel).GetAwaiter().GetResult();
+		}
+
+		private async Task<DbTransaction> BeginDbTransactionAsync(IsolationLevel isolationLevel, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
Can the casts above be avoided if this returns `Task<MySqlTransaction>`? Since it's a private method, I see no reason to not return the most-derived type.